language quickstart review (#6778)
circleci/circleci-docs: jekyll/_cci2/language-javascript.md

---

{% include snippets/language-guided-tour-cards.md lang="Node.JS" demo_url_slug="javascript" demo_branch="master" guide_completion_time="15" sample_completion_time="10" %}

## Overview
{: #overview }

This is a quickstart guide for integrating a Node.JS project with CircleCI. This guide is designed to help you create a basic CircleCI configuration file to build, test and deploy your Node.JS project. After completing this quickstart you can edit and optimize the config to fit the requirements of your project.

## Prerequisites
{: #prerequisites}

* [A CircleCI account]({{site.baseurl}}/2.0/first-steps/)
* A Node.JS project located in a supported VCS (currently GitHub or Bitbucket)

If you do not have a Node.JS project, but would like to follow this guide, you can use our sample project, which is [hosted on GitHub]({{site.gh_public_org_url}}/sample-javascript-cfd)
and is [building on CircleCI]({{site.cci_public_org_url}}/sample-javascript-cfd). Consider [forking the repository]({{site.gh_help_articles_url}}/fork-a-repo/)
and rewriting [the configuration file]({{site.gh_public_org_url}}/sample-javascript-cfd/blob/master/.circleci/config.yml)
as you follow this guide.

> ## Configuration Walkthrough

Author: ```suggestion
## Configuration walkthrough
```

To add the orb to your config, insert:
```yaml
orbs:
  node: circleci/node@5.0.2
```

> **Note**: It is always a good idea to check in the [Orb Registry](https://circleci.com/developer/orbs) to ensure you are using the most recent (or correct for your project) orb verison.

Author: ```suggestion
**Note**: When using an orb, it is a good idea to check the [Orb Registry](https://circleci.com/developer/orbs) to ensure you are using the most recent version, or the version that fits best with your specific project.
```

### 5. Conclusion
{: #conclusion }

You just set up a Node.js app to build on CircleCI. Check out your project’s [pipeline page]({{site.baseurl}}/2.0/project-build/#overview) to see how this looks when building on CircleCI.

> ## Full configuration file

Author: Currently the javascript project we link to has a `master` branch rather than `main` but I have requested this be updated for consistency.
